libsolidity/ast/Types.cpp

```diff
@@ -2565,6 +2565,8 @@ u256 FunctionType::externalIdentifier() const
 
 bool FunctionType::isPure() const
 {
+	// FIXME: replace this with m_stateMutability == StateMutability::Pure once
```

Comment:
note that `isPure` is more strict than the pure annotation. We should probably rename this function

Reply:
What is it exactly then? The documentation in the header doesn't suggest it would be more strict (assuming callgraph detects all violations).

Reply:
The entire comment is:
```
       // FIXME: replace this with m_stateMutability == StateMutability::Pure once
       //        the callgraph analyzer is in place
```

Reply:
Sorry, it is at least used in a more restricted way. The result if this function is basically assigned to ExpressionAnnotation::isPure, which is used to determine if the value is a compile-time constant. If also allow functions with `pure` modifier here, it is not compile-time constant anymore - or is it? (I was thinking about function arguments which can be modified by pure functions)

Reply:
A compile time constant such as `uint constant a = sha3(1) * 5;`? In think any pure function satisfies that as long as the parameters to it are constant.